WordPress/src/main/java/org/wordpress/android/util/analytics/service/InstallationReferrerServiceLogic.java

```diff
@@ -0,0 +1,127 @@
+            @Override
+            public void onInstallReferrerSetupFinished(int responseCode) {
+                switch (responseCode) {
+                    case InstallReferrerResponse.OK:
+                        // Connection established
+                        try {
+                            AppLog.i(T.UTILS, "installation referrer connected");
+                            ReferrerDetails response = mReferrerClient.getInstallReferrer();
+                            // mark referrer as obtained so we don't try fetching it again
+                            // Citing here:
+                            //  Caution: The install referrer information will be available for 90 days and won't
+                            //  change unless the application is reinstalled. To avoid unecessary API calls in your
+                            //  app, you should invoke the API only once during the first execution after install.
+                            // read more: https://developer.android.com/google/play/installreferrer/library
+                            AppPrefs.setInstallationReferrerObtained(true);
+
+                            // handle and send information to Tracks here
+                            Map<String, Object> properties = new HashMap<>();
+                            properties.put("install_referrer", response.getInstallReferrer());
+                            properties.put("install_referrer_timestamp_begin",
+                                    response.getInstallBeginTimestampSeconds());
+                            properties.put("install_referrer_timestamp_click",
+                                    response.getReferrerClickTimestampSeconds());
+                            AnalyticsTracker.track(AnalyticsTracker.Stat.INSTALLATION_REFERRER_OBTAINED, properties);
+                            mReferrerClient.endConnection();
+                        } catch (RemoteException e) {
+                            e.printStackTrace();
+                            AppLog.i(T.UTILS, "installation referrer: RemoteException occurred");
+                        }
+                        break;
+                    case InstallReferrerResponse.FEATURE_NOT_SUPPORTED:
+                        // API not available on the current Play Store app
+                        // just log but DO NOT mark AppPrefs.setInstallationReferrerObtained(true);,
+                        // because the user could update to a version of Google PLay that supports it
+                        // and we can obtain it also from the old com.android.vending.INSTALL_REFERRER intent
+                        AppLog.i(T.UTILS, "installation referrer: feature not supported");
+                        break;
+                    case InstallReferrerResponse.SERVICE_UNAVAILABLE:
+                        // Connection could not be established
+                        // same as above, this is a retriable error
```

Comment:
Can we change `this is a retriable error` to `this error can be retried`?

Reply:
addressed in 0cd3efc5b3